pkg/cluster/delete.go

```diff
@@ -259,9 +259,20 @@ func (m *manager) deleteRoleDefinition(ctx context.Context) error {
 
 func (m *manager) Delete(ctx context.Context) error {
 	resourceGroup := stringutils.LastTokenByte(m.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID, '/')
+	rg, err := m.resourceGroups.Get(ctx, resourceGroup)
+	if err != nil {
+		return err
+	}
+	//In edge case of CRG not being managedBy ARO, we have a different delete path
+	//we will assume normal case and set rgManagedByARO to true CRG is managedby ARO
+	rgManagedByARO := true
+	if rg.ManagedBy == nil || *rg.ManagedBy == "" || !strings.EqualFold(*rg.ManagedBy, m.doc.OpenShiftCluster.ID) {
+		rgManagedByARO = false
+		m.log.Infof("cluster resource group not managed by aro %s", *rg.Name)
+	}
```

Comment:
Do something like this:
```
//In edge case of CRG not being managedBy ARO, we have a different delete path
//we will assume normal case and set rgManagedByARO to true CRG is managedby ARO
rgManagedByARO := true
rg, err := m.resourceGroups.Get(ctx, resourceGroup)
if err != nil {
	m.log.Warn("failed to get resourceGroup %s", err)
} else {
        if rg.ManagedBy == nil || *rg.ManagedBy == "" || !strings.EqualFold(*rg.ManagedBy, m.doc.OpenShiftCluster.ID) 
        {
	       rgManagedByARO = false
	       m.log.Infof("cluster resource group not managed by aro %s", *rg.Name)
         }
}
```
I think we should not fail if we failed to get resource group and still attempt to delete. This might be due to missing permissions of flakes

Reply:
done